### PR TITLE
Wrong HTTPS example fixed. Will save others my #5734 hours of debugging

### PR DESCRIPTION
**Pull request by joysfera**
The provided HTTPS examples are misleading and plain wrong because don't ensure the client object lives longer than the https object, which is required per comment hidden in the source code. Otherwise nasty crashes will occur (#5734 ). I have fixed it using a nested code block and a clear comment.

### CODE REVIEW

File: libraries/ESP8266HTTPClient/examples/BasicHttpsClient/BasicHttpsClient.ino [hunk 1]
```diff
@@ -45,32 +45,36 @@ void loop() {
...
-        // HTTP header has been send and Server response header has been handled
-        Serial.printf("[HTTPS] GET... code: %d\n", httpCode);
-
-        // file found at server
-        if (httpCode == HTTP_CODE_OK || httpCode == HTTP_CODE_MOVED_PERMANENTLY) {
-          String payload = https.getString();
-          Serial.println(payload);
+    /* in order to ensure the client object lives the entire time of the HTTPClient
+       the HTTPClient is in its own nested code block */
+    {
```
Comment (devyte) [inline libraries/ESP8266HTTPClient/examples/BasicHttpsClient/BasicHttpsClient.ino]: HTTPClient doesn't need to be in its own code block, it just needs to be instantiated _after_ client. Destructors are called in reverse order, which ensures https gets destroyed first and client second.
I suggest removing the code block, and updating the comment to explain this.

File: libraries/ESP8266HTTPClient/examples/StreamHttpsClient/StreamHttpsClient.ino [hunk 2]
```diff
@@ -54,56 +54,60 @@ void loop() {
 
     client->setFingerprint(fingerprint);
 
-    HTTPClient https;
+    /* in order to ensure the client object lives the entire time of the HTTPClient
+       the HTTPClient is in its own nested code block */
+    {
```
Comment (devyte) [inline libraries/ESP8266HTTPClient/examples/StreamHttpsClient/StreamHttpsClient.ino]: Same as above.

File: libraries/ESP8266HTTPClient/examples/BasicHttpsClient/BasicHttpsClient.ino (hunk 1 above)
Comment (joysfera) [reply libraries/ESP8266HTTPClient/examples/BasicHttpsClient/BasicHttpsClient.ino]: I don't agree. It's way too dangerous to rely on order of code lines. Took me two days to debug that. It should be totally explicit, IMHO.

**Comment by d-a-v** [comment]
@joysfera The nested block is not the preferred way but the information or warning is relevant. 
Would you update your PR with a comment to inform readers about the requirement ?
Also the last debug line is interesting to have.

**Comment by joysfera** [comment]
I understand you don't want to accept the nested block solution. However, I don't agree with a "solution" based on the order of variable declarations. Adding a comment doesn't really "fix" that.
The whole API is wrong. Possible solution is listed here: https://github.com/esp8266/Arduino/issues/5734#issuecomment-475858786

**Comment by earlephilhower** [comment]
Closing as abandoned with changes requested.

**Pull request closed without merging**